Drive/API/DriveService.swift

```diff
@@ -61,24 +60,49 @@ class DriveService: DriveServiceType {
                 return completion(.failure(.invalidResponse))
             }
 
-            guard let registerResponse = try? JSONDecoder().decode(RegisterResponse.self, from: data) else {
+            guard let putUserResponse = try? JSONDecoder().decode(PutUserResponse.self, from: data) else {
                 return completion(.failure(.invalidResponse))
             }
 
-            let user = User(registerResponse: registerResponse)
+            let user = User(putUserResponse: putUserResponse)
             completion(.success(user))
         }
 
         dataTask.resume()
     }
+    
+    // MARK: - Get Drives
+    
+    static func getDrives(completion: @escaping GetDrivesCompletion) {
+        guard let request = RequestFactory.urlRequest(for: Endpoint.getDrives) else {
+            completion(.failure(.invalidRequest))
+            return
+        }
+
+        let dataTask = urlSession.dataTask(with: request) { data, _, _ in
+            guard let data = data else {
+                return completion(.failure(.invalidResponse))
+            }
+
+            guard let allDrivesResponse = try? JSONDecoder().decode(GetDrivesResponse.self, from: data) else {
+                return completion(.failure(.invalidResponse))
+            }
+            
+            completion(.success(allDrivesResponse))
+        }
+
+        dataTask.resume()
+    }
+    
+    // MARK: - Save Drive
 
-    func saveDrive(session: DrivingSession, completion: @escaping SaveDriveCompletion) {
-        guard let user = user else {
+    static func saveDrive(session: DrivingSession, completion: @escaping SaveDriveCompletion) {
+        guard let user = UserProvider.shared.currentUser else {
             completion(.failure(.invalidCurrentUser))
             return
         }
 
-        let endpoint = Endpoint.drive(user: user, session: session)
+        let endpoint = Endpoint.putDrive(user: user, session: session)
```

Comment:
Damn yeah I see now that the `Endpoint.drive` was a bit ambiguous. Now it's easy to know exactly what the endpoint does 💯